library/Zend/Mvc/Controller/AbstractRestfulController.php

```diff
@@ -374,8 +375,12 @@ public function onDispatch(MvcEvent $e)
                     $id = null;
                 }
                 $action = 'head';
-                $this->head($id);
+                $headResult = $this->head($id);
                 $response = $e->getResponse();
+                if ($headResult instanceof HttpResponse) {
+                    $response->setStatusCode($headResult->getStatusCode());
```

Comment:
@Ocramius I've updated the code. let me know if this solution is a least a step in the right direction.